filters.go

```diff
@@ -109,6 +109,46 @@ func (p *Pipe) First(lines int) *Pipe {
 	return Echo(output.String())
 }
 
+// Last reads from the pipe, and returns a new pipe containing only the last N
+// lines. If there is an error reading the pipe, the pipe's error status is also
+// set.
+func (p *Pipe) Last(lines int) *Pipe {
+	if p == nil || p.Error() != nil {
+		return p
+	}
+	scanner := bufio.NewScanner(p.Reader)
+	var stringArr []string
+	for scanner.Scan(){
+		stringArr = append(stringArr, scanner.Text())
```

Comment:
The problem with this is that we read the _whole_ pipe into memory, before throwing away most of it. Would it be more efficient to use something like a ring buffer that just stores the last X lines?

Reply:
I can't think of a more efficient way than to read the whole pipe and the current implementation exposes a Reader, and the Reader interface does not have functions to find the length of the content or going to an arbitrary token index.
That is why to find the last n tokens, we need the number of tokens and for that, we need to read the whole Pipe.

Please let me know if I am wrong or if there is a more efficient method to read the content.

Reply:
Well, to take a trivial example, suppose we just want the very last line. We could do something like the following:

1. Read a line from the input
2. If not the end of input, goto 1.
3. Return the line

The question is really how many lines we need to store in memory, given that we want the last X lines. The logical answer is just X, isn't it?

Reply:
Yes, will try to incorporate this logic also